src/Components/Web/src/Forms/InputRadioGroup.cs

```diff
@@ -43,11 +43,15 @@ protected override void BuildRenderTree(RenderTreeBuilder builder)
         {
             Debug.Assert(_context != null);
 
-            builder.OpenComponent<CascadingValue<InputRadioContext>>(2);
-            builder.AddAttribute(3, "IsFixed", true);
-            builder.AddAttribute(4, "Value", _context);
-            builder.AddAttribute(5, "ChildContent", ChildContent);
+            builder.OpenRegion(_context.GetHashCode());
```

Comment:
This is an interesting solution! I agree it's good to let the `IsFixed` parameter stay but instead cause the child subtree to reinitialize if the context changes.

However, could I suggest that instead of using the region with the hashcode sequence number, we eliminate the region and put a `key` on the `CascadingValue<InputRadioContext>` instead? For example, by calling `builder.SetKey(_context);` just before or after its attributes are set.

If you already tried that and it doesn't work for some reason, then perhaps the region is the best approach. However if the key is viable, I'd say it's preferable because:

 * This is exactly what `key` is supposed to be for, so there's less magic going on and less need to understand diffing internals
 * It doesn't set a pattern of dynamic sequence numbers that have to be justified on a case-by-case basis
 * Although it's incredibly improbably, it's technically possible for two different `_context` values to have the same hash code just by chance. I know it's literally one-in-4-billion but hey maybe someone somewhere will encounter an unreproducible bug :)